docs/development_falcon.rst

```diff
@@ -0,0 +1,19 @@
+Falcon Development
+==================
+
+Naming conventions
+------------------
+Custom modules should be placed into the `modules/custom` directory and
+prefixed with `falcon_` string, for instance `falcon_example`.
+
+Packaged features should be placed into `modules/features` directory and
+prefixed with `falcon_feature_` string, for instance `falcon_feature_example`.
+
+Adding a new module
+-------------------
+When adding a new module to the Falcon distribution developer should decide if
+it is needed on every site consuming the distribution. If this is the case then
+it should be enabled using hook_update_N() in the `falcon_deploy` module.
```

Comment:
Shall it be added to `core.extensions.yml` or `falcon.info.yml`  as well?

Reply:
Good catch! Added info about adding it as a dependency into falcon.info.yml file.